crates/swift-bridge-build/src/package.rs

```diff
@@ -9,23 +9,23 @@ use std::process::Command;
 use tempfile::tempdir;
 
 /// Config for generating Swift packages
-pub struct CreatePackageConfig<'a> {
+pub struct CreatePackageConfig<'a, P: AsRef<Path> + ?Sized> {
     /// The directory containing the generated bridges
-    pub bridge_dir: &'a dyn AsRef<Path>,
+    pub bridge_dir: &'a P,
     /// Path per platform. e.g. `(ApplePlatform::iOS, "target/aarch64-apple-ios/debug/libmy_rust_lib.a")`
-    pub paths: HashMap<ApplePlatform, &'a dyn AsRef<Path>>,
+    pub paths: HashMap<ApplePlatform, &'a P>,
     /// The directory where the package will be saved
-    pub out_dir: &'a dyn AsRef<Path>,
+    pub out_dir: &'a P,
     /// The name for the Swift package
     pub package_name: &'a str,
 }
 
-impl<'a> CreatePackageConfig<'a> {
+impl<'a, P: AsRef<Path> + ?Sized> CreatePackageConfig<'a, P> {
```

Comment:
We don't want this generic bounds because it constraints every field to use the same monomorphized `P`